packages/vm/src/checksum.rs

```diff
@@ -19,7 +20,16 @@ impl Checksum {
 
     /// Creates a lowercase hex encoded copy of this checksum
     pub fn to_hex(&self) -> String {
-        hex::encode(self.0)
+        self.to_string()
+    }
+}
+
+impl fmt::Display for Checksum {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        for byte in self.0.iter() {
+            write!(f, "{:02x}", byte)?;
```

Comment:
Why not something like `write!(f, "{}", self.to_hex())` ?

Reply:
The idea is that we don't need the external `hex` crate when using format printing like this.

Reply:
Sure. But with what I'm proposing, you don't need it either.
I'm suggesting to use basically `to_string()` (i.e `Display` impl) here too. Which is in fact exactly similar to what you did.
Only that without replicating the code.

Reply:
Hmm, I guess I don't understand. Which duplication? `to_hex()` uses `to_string()` which is auto-implemented via `Display` using the `fmt` implementation from here.

Reply:
I see, the chain is the other way around. Sorry, I got confused, there's no code duplication, and implementation is good as it is.

Reply:
Alright. I did it in this direction to avoid a heap String in cases where we don't need one.